client/lib/signup/step-actions.js

```diff
@@ -613,10 +613,15 @@ export function isSiteTypeFulfilled( stepName, defaultDependencies, nextProps )
 		initialContext: {
 			query: { site_type: siteType },
 		},
+		signupDependencies,
 	} = nextProps;
 	const siteTypeValue = getSiteTypePropertyValue( 'slug', siteType, 'slug' );
 	let fulfilledDependencies = [];
 
+	if ( siteType === get( signupDependencies, 'siteType' ) ) {
```

Comment:
Should this be `siteTypeValue ===  `? 
Also, could you help me understand in which scenario this condition is fulfilled?

Reply:
@niranjan-uma-shankar We store `siteType` in the next 6 lines. I guess `siteTypeValue` is used for detecting if `siteType` is valid.